code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -92,62 +85,110 @@ func (zcn *ZCNSmartContract) AddAuthorizer(tran *transaction.Transaction, inputD
 		return "", err
 	}
 
-	// compare the global min of authorizerNode Authorizer to that of the transaction amount
-	if globalNode.MinStakeAmount > state.Balance(tran.Value*1e10) {
-		msg := fmt.Sprintf("min stake amount '(%d)' > transaction value '(%d)'", globalNode.MinStakeAmount, tran.Value)
-		err = common.NewError(code, msg)
-		Logger.Error("min stake amount > transaction value", zap.Error(err))
-		return "", err
-	}
-
-	// Create Authorizer instance
-
-	authorizer = NewAuthorizer(authorizerID, authorizerPublicKey, authorizerURL)
-
-	// Dig pool for authorizer
-
-	transfer, response, err := authorizer.Staking.DigPool(tran.Hash, tran)
-	if err != nil {
-		err = common.NewError(code, fmt.Sprintf("error digging pool, err: (%v)", err))
-		return "", err
-	}
-
-	err = ctx.AddTransfer(transfer)
-	if err != nil {
-		msg := "Error: '%v', transaction.ClientId: '%s', transaction.ToClientId: '%s', transfer.ClientID: '%s', transfer.ToClientID: '%s'"
-		err = common.NewError(
-			code,
-			fmt.Sprintf(
-				msg,
-				err,
-				authorizerID,
-				recipientID,
-				transfer.ClientID,
-				transfer.ToClientID,
-			),
-		)
-		return "", err
+	// create stake pool for the validator to count its rewards'
+
+	var createOrUpdateStakePool = func() (err error) {
+		var sp *StakePool
+		sp, err = zcn.getOrUpdateStakePool(globalNode, authorizerID, spenum.Authorizer, authorizerStakingPoolSettings, ctx)
+		if err != nil {
+			return common.NewError(code, "failed to get or create stake pool: "+err.Error())
+		}
+		if err = sp.save(zcn.ID, authorizerID, ctx); err != nil {
+			return common.NewError(code, "failed to save stake pool: "+err.Error())
+		}
+		return err
 	}
 
-	err = authorizer.Save(ctx)
-	if err != nil {
-		msg := fmt.Sprintf("error saving authorizer(authorizerID: %v), err: %v", authorizerID, err)
-		err = common.NewError(code, msg)
-		Logger.Error("saving authorizer node", zap.Error(err))
-		return "", err
-	}
+	// Check existing Authorizer
 
-	ev, err := authorizer.ToEvent()
-	if err != nil {
-		msg := fmt.Sprintf("error marshalling authorizer(authorizerID: %v) to event, err: %v", authorizerID, err)
-		err = common.NewError(code, msg)
-		Logger.Error("emitting event", zap.Error(err))
-		return "", err
+	authorizer, err = GetAuthorizerNode(authorizerID, ctx)
+	if err == nil && authorizer != nil {
+		errs := fmt.Errorf("authorizer(authorizerID: %v) already exists", authorizerID)
+		err = createOrUpdateStakePool()
+		if err != nil {
+			errs = multierror.Append(errs, errors.Wrap(err, "failed to get or create stake pool"))
+		} else {
+			errs = errors.Wrap(errs, "create or update stake pool completed successfully")
```

Comment:
We don't need to add success msg to errors

Reply:
Ok